src/frequenz/sdk/microgrid/component_graph.py

```diff
@@ -289,8 +291,31 @@ def dfs(
             the condition function.
         """
 
+    @abstractmethod
+    def find_first_descendant(
+        self,
+        root_category: ComponentCategory = ComponentCategory.GRID,
+        descendant_categories: Iterable[ComponentCategory] | None = None,
```

Comment:
I really don't see the point of providing defaults here. With such a generic name it doesn't say much to call `find_first_decendant()`, decendant of what?

Also I don't think there is a reasonable default for `decendant_categories`. This is a very low level helper function, is not something that people will have to call all the time, so it should be fine to make it explicit.

```suggestion
        *,
        root_category: ComponentCategory,
        descendant_categories: Iterable[ComponentCategory],
```

Reply:
Amended as suggested